Mathlib/LinearAlgebra/AffineSpace/Pointwise.lean

```diff
@@ -72,4 +74,54 @@ theorem map_pointwise_vadd (f : P₁ →ᵃ[k] P₂) (v : V₁) (s : AffineSubsp
   ext
   exact f.map_vadd _ _
 
+section SMul
+variable [Monoid M] [DistribMulAction M V] [SMulCommClass M k V] {a : M}
+
+/-- The multiplicative action on an affine subspace corresponding to applying the action to every
+element.
+
+This is available as an instance in the `Pointwise` locale. -/
+protected def pointwiseSMul : SMul M (AffineSubspace k V) where
+  smul a s := s.map (DistribMulAction.toLinearMap k _ a).toAffineMap
+
+scoped[Pointwise] attribute [instance] AffineSubspace.pointwiseSMul
+
+@[simp, norm_cast]
+lemma coe_smul (a : M) (s : AffineSubspace k V) : ↑(a • s) = a • (s : Set V) := rfl
+
+/-- The multiplicative action on an affine subspace corresponding to applying the action to every
+element.
+
+This is available as an instance in the `Pointwise` locale. -/
+protected def mulAction : MulAction M (AffineSubspace k V) :=
+  SetLike.coe_injective.mulAction _ coe_smul
+
+scoped[Pointwise] attribute [instance] AffineSubspace.mulAction
+
+lemma smul_eq_map (a : M) (s : AffineSubspace k V) :
+    a • s = s.map (DistribMulAction.toLinearMap k _ a).toAffineMap := rfl
+
+lemma smul_mem_smul_iff (ha : IsUnit a) {s : AffineSubspace k V} {p : V} :
+    a • p ∈ a • s ↔ p ∈ s := smul_mem_smul_set_iff (a := ha.unit)
```

Comment:
I think elsewhere we state this kind of thing separately for `Group` and `IsUnit`? (proving the latter from the former)

Reply:
Well actually all the examples that come to mind only have a `Group` and `GroupWithZero` versions, and are missing the `IsUnit` version? eg `Set.smul_mem_smul_set_iff`/`Set.smul_mem_smul_set_iff₀`

Reply:
It might be worth being consistent here, especially if you don't care about the `IsUnit` case. Alternatively, perhaps leave a TODO about copying this generality back to the other lemmas?

Reply:
Done